17677: update firewall_instance_association read

CyrusJavan opened the pull request:
Update firewall_instance_association read function
to handle the special case of Azure FQDN gateway
that has a lan_interface.

arvindsree commented:
Update the docs as well to remove empty lists for blocks

arvindsree commented on aviatrix/resource_aviatrix_fqdn.go:
```diff
@@ -45,7 +45,7 @@ func resourceAviatrixFQDN() *schema.Resource {
+							Required:    true,
 							Description: "Name of the gateway to attach to the specific tag.",
 						},
 						"source_ip_list": {
```
Should we give default value for this?

zhiqiangjin replied:
No, we should give the name of the gateway. I think it should be required.

arvindsree replied:
i meant for source IP list

arvindsree replied:
default as empty list? just test it if it is required or it takes default as null

zhiqiangjin replied:
It takes default as null. We can leave it as optional.

arvindsree replied:
will code crash for null?

CyrusJavan committed:
17677: update firewall_instance_association read

Update firewall_instance_association read function
to handle the special case of Azure FQDN gateway
that has a lan_interface.
